extensions/circuit/src/adapters/allocate_frame.rs

```diff
@@ -76,19 +76,26 @@ pub struct AllocateFrameWriteRecord<F> {
 pub struct AllocateFrameAdapterColsWom<T> {
     pub from_state: ExecutionState<T>,
     pub from_frame: FrameState<T>,
-    pub target_reg_ptr: T,  // Pointer to target register
-    pub allocation_size: T, // amount_imm field from instruction (b)
-    pub src_reg_ptr: T,     // if reading from register
-    pub result: T,
-    pub read_imm_or_reg: T,
+    pub amount_reg: T,
+    // amount from register
+    pub amount: [T; RV32_REGISTER_NUM_LIMBS],
+    // immediate amount
+    pub amount_imm: T,
```

Comment:
Isn't this the same as `allocation_size`?

Reply:
what I meant is that there are 3 values in columns now, the reg value and 2 immediates, so it feels like the immediates are duplicated